driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java

```diff
@@ -48,7 +49,8 @@
 
     private static final RootCodecRegistry DEFAULT_CODEC_REGISTRY = new RootCodecRegistry(asList(new ValueCodecProvider(),
                                                                                                  new DocumentCodecProvider(),
-                                                                                                 new BsonValueCodecProvider()));
+                                                                                                 new BsonValueCodecProvider(),
+                                                                                                 new GridFSFileCodecProvider()));
```

Comment:
As of JAVA-1596: Now adding the codec to the files collection/database which is used by the gridfs instance. Reverted any changes to the CodecRegistry